code/modules/library/book.dm

```diff
@@ -0,0 +1,370 @@
+				return
+			if(length(pages) >= MAX_PAGES)
+				to_chat(usr, "<span class='notice'>You can't fit anymore pages in this book!</span>")
+				return
+			to_chat(usr, "<span class='notice'>You add another page to the book!</span>")
+			pages += " "
+		if("Remove Page")
+			if(!length(pages))
+				to_chat(usr, "<span class='notice'>There aren't any pages in this book!</span>")
+				return
+			var/page_choice = stripped_input(usr, "There are [length(pages)] pages, which page number would you like to remove?")
+			if(!isnum(page_choice) || page_choice > length(pages))
+				to_chat(usr, "That is not an acceptable value.")
+				return
+			pages -= pages[page_choice]
+
+/obj/item/book/proc/edit_page(content, page_number)
+	if(!content)
+		return
+	if(length(pages) < page_number) //can't remove the page if it doesn't exist
+		return
+	pages += content
+
+/obj/item/book/proc/add_page(content)
+	var/text = content
+	if(!text)
+		text = " "
+	if(length(pages) >= MAX_PAGES)
+		return FALSE
+	pages += text
+	current_page = length(pages) //open to newely added page so player can edit it
+	return TRUE
+
+/obj/item/book/proc/remove_page(page_number)
+	if(length(pages) < page_number) //can't remove the page if it doesn't exist
+		return FALSE
+	if(pages -= pages[page_number])
+		page_number = length(pages) //if page_number is somehow at a value it shouldn't be we fix it here aswell
+		return TRUE //we want to make sure whatever is calling this proc knows the operation was succesful
+	return FALSE
+
+/obj/item/book/proc/carve_book(mob/user, obj/item/I)
+	if(carved)
+		to_chat(user, "<span class='warning>[title] has already been carved out!</span>")
+		return
+	if(!I.sharp)
+		to_chat(user, "<span class='warning>You can't carve [title] using that!</span>")
+		return
+	to_chat(user, "<span class='notice'>You begin to carve out [title].</span>")
+	if(I.use_tool(src, user, 30, volume = I.tool_volume))
+		user.visible_message("<span class='warning'>[user] appears to carve out the pages inside of [title]!</span>",\
+				"<span class='danger'>You begin to carve out [title].!</span>")
+		carved = TRUE
+		return TRUE
+
+/obj/item/book/proc/store_item(obj/item/I, mob/user)
+	if(store)
+		to_chat(user, "<span class='notice'>There is already something in [name]!</span>")
+		return
+
+	//does it exist, if so is it an abstract item?
+	if(!I || !istype(I) || (I.flags & ABSTRACT))
+		return
+	if(I.flags & NODROP)
+		to_chat(user, "<span class='notice'>[I] stays stuck to your hand when you try and hide it in the book!.</span>")
+	//Checking to make sure the item we're storing isn't larger than/equal to size of the book, prevents recursive storing aswell
+	if(I.w_class >= w_class)
+		to_chat(user, "<span class='notice'>[I] is to large to fit in [name].</span>")
+		return
+
+	user.drop_item()
+	I.forceMove(src)
+	store = I
+	to_chat(user, "<span class='notice'>You hide [I] in [name].</span>")
+
+
+/obj/item/book/proc/remove_stored_item(mob/user, thrown = FALSE)
```

Comment:
```suggestion
/obj/item/book/proc/remove_stored_item(mob/user, display_message = FALSE)
```
Would make it more clear what the variable does. Also means you don't have to negate it when checking

Reply:
added

Reply:
added